packages/admin/admin-rte/src/core/xml/translateAndTransformXmlToState.ts

```diff
@@ -0,0 +1,198 @@
+import {
+    ContentState,
+    convertFromRaw,
+    DraftInlineStyleType,
+    EditorState,
+    RawDraftContentBlock,
+    RawDraftContentState,
+    RawDraftEntity,
+} from "draft-js";
+
+import { blockToXml } from "./transformStateToXml";
+
+interface InlineStyle {
+    id: number;
+    offset: number;
+    length: number;
+}
+
+interface EntityRange {
+    key: number;
+    offset: number;
+    length: number;
+}
+
+export const updateBlockContent = (block: RawDraftContentBlock) => {
+    const regexInlineStylesPattern = /<inline id="[0-9][0-9]?">|<\/inline>/g;
+    const regexEntitiesPattern = /<entity id="[0-9][0-9]?">|<\/entity>/g;
+
+    const newInlineStyleRanges: InlineStyle[] = [];
+    const newEntityRanges: EntityRange[] = [];
+    const stylesStack: InlineStyle[] = [];
+    const entitiesStack: EntityRange[] = [];
+
+    // process only inline style ranges and their associated pseudo tags
+    const onlyInlineStyles = block.text.replace(regexEntitiesPattern, "");
+    const inlineStyleTags = onlyInlineStyles.matchAll(regexInlineStylesPattern);
+
+    // process only entity ranges and their associated pseudo tags
+    const onlyEntityRanges = block.text.replace(regexInlineStylesPattern, "");
+    const entityRangesTags = onlyEntityRanges.matchAll(regexEntitiesPattern);
+
+    const text = block.text.replace(regexEntitiesPattern, "").replace(regexInlineStylesPattern, "");
+
+    // counts the number of characters of previous pseudo tags to keep the correct index shift
+    let shiftCount = 0;
+
+    for (const match of inlineStyleTags) {
+        const id = match[0].match(/\d+/)?.[0]; // get id of opening pseudo tag
+
+        if (match.index === null || match.index === undefined) continue;
+
+        const offset = match.index - shiftCount;
+
+        if (id) {
+            // opening pseudo tags are added to the stack
+            stylesStack.push({
+                id: parseInt(id ?? ""),
+                offset,
+                length: 0,
+            });
+        } else {
+            // take top element of stack when closing pseudo tags occurs
+            const openingTag = stylesStack.pop();
+
+            if (openingTag) {
+                // check if style already occured before
+                const existingStyle = newInlineStyleRanges.findIndex((style) => style.id === openingTag.id);
```

Comment:
When could this happen? When a style is interrupted by an entity?

Reply:
yes, for example if you have the following text: 
"This is an example **bold [link](https://www.google.com) to google**"
then the XML would look like this:

`This is an example <inline id="1">bold </inline><entity id="1"><inline id="1">link</inline></entity><inline id="1"> to google</inline>`